### ISSUE TITLE
wifi: Add support for OWE (WFA Enhanced open)

### ISSUE THREAD

**Issue by krish2718** [zephyrproject-rtos/zephyr#77027]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Add support for https://en.wikipedia.org/wiki/Opportunistic_Wireless_Encryption which is used in few deployment under the WFA name `Enhanced Open`

**Describe alternatives you've considered**

**Additional context**

**Comment by antonrejoe** [comment]
Update: Testing and Debugging OWE support using ESP32 Wrover-E module, will submit a PR soon.

**Comment by antonrejoe** [comment]
@krish2718 What would you suggest for creating an OWE AP? I tried creating a Soft AP using hostapd (in Endeavour OS), but I am unable to do it. And on further investigation it seems like, my firmware does not support it. Thanks in Advance !!

**Comment by krish2718** [comment]
> [@krish2718](https://github.com/krish2718) What would you suggest for creating an OWE AP? I tried creating a Soft AP using hostapd (in Endeavour OS), but I am unable to do it. And on further investigation it seems like, my firmware does not support it. Thanks in Advance !!

SoftAP mode doesn't support these advanced security methods. You can try using full `hostapd` based AP, NXP chips support this mode and they might have OWE support too. But best bet is to get a COTS AP that supports OWE rather than using ZephyrRTOS as AP.

**Comment by antonrejoe** [comment]
@krish2718  Thank you!! Will update you on this.

**Comment by antonrejoe** [comment]
Update: @krish2718 I tried to get the hardware, but my attempt was futile. May I submit a PR?

**Comment by krish2718** [comment]
> Update: [@krish2718](https://github.com/krish2718) I tried to get the hardware, but my attempt was futile. May I submit a PR?

Yes, please. Let's see if we can find some volunteers for testing on real HW.
